ArtOfIllusion/src/artofillusion/ViewerCanvas.java

```diff
@@ -692,14 +692,17 @@ public void setNavigationMode(int nextNavigation, boolean nextPerspective)
       return;
     }
 
-    
-	// Change perspective without animation, if needed.
-    if (nextNavigation < 2) // ...?
-	  perspective = nextPerspective;
-    else
-	  perspective = true;
-	  
-    // Turn y up for landscape modes. Animated
+    // Change perspective without animation. This is a bit blunt but softer  
+    // options would need a more complex set of conditions.
+
+    if (nextNavigation < 2)
+        flipPerspectiveSwitch(nextPerspective);
+    else{
```

Comment:
Hey, missed the closing brace here...

Reply:
Sorry. That '{' should have been removed, I had a debugging tag there... Must have deviated from my usual work flow, this would have been caught at compile.